public/Leo Jayachandran/Glossarizer/libretextsglossarizer.js

```diff
@@ -0,0 +1,629 @@
+                    target.bind('mouseleave', function () {
+                        setTimeout(function (word) {
+                            if ($(`#tooltip${word}:hover`).length == 0) {
+                                $("#tooltip" + word).animate({
+                                    top: '-=10',
+                                    opacity: 0
+                                }, 50, function () {
+                                    $(`.glossarizer_replaced:contains('${word}')`).attr("title", $(this).html())
+                                    $(this).remove();
+                                });
+                            }
+                        }, 300, target.html())
+                    });
+                });
+
+            }
+
+            return window.tooltip = ToolTip;
+
+        })(jQuery, window)
+
+        //Initialise Glossariser
+        $(function () {
+            $('.mt-content-container').glossarizer({
+                callback: function () {
+                    new tooltip();
+                }
+            });
+
+
+        });
+        /**
+         * Defaults
+         */
+
+    },
+    removeGlossary: function () {
+        $('.mt-content-container').removeData('plugin_glossarizer');
+
+        /* Remove wrapping tag */
+        $('.mt-content-container').find('.glossarizer_replaced').each(function () {
+            let $this = $(this),
+                text = $this.text()
+
+            $this.replaceWith(text);
+        });
+    },
+    buildBackMatter: function () {
+        let $glossaryTable = $("table:contains('Word" + "(s)')")
+        let tBody = $glossaryTable.html().replace(/&nbsp;/g, " ").replace(/<p>/g, " ").replace(/<\/p>/g, " ").trim();
+        tBody = tBody.substring(tBody.search("<tbody"), tBody.search("</tbody>")).trim();
+        let tableRows = [];
+        for (let i = 0; i < tBody.length;) {
+            let trimmedBody = tBody.substring(i);
+            let rowStart = '<tr>';
+            let rowEnd = '</tr>';
+            let rowContent = trimmedBody.substring(trimmedBody.search(rowStart) + rowStart.length, trimmedBody.search(rowEnd)).trim();
+            tableRows.push(rowContent);
+            i += trimmedBody.search(rowEnd) + rowEnd.length;
+        }
+        //Generate the Glossary
+        let glossaryList = [];
+        for (let r = 0; r < tableRows.length; r++) {
+            let newTerm = {
+                "term": "",
+                "description": ""
+            };
+            //Get cells in the 3 columns
+            let cols = {};
+            let colStart = [
+                ['<td data-th="Word' + '(s)">', "word"],
+                ['<td data-th="Definition">', "definition"],
+                ['<td data-th="Exclusions">', "exclusion"],
+                ['<td data-th="Image">', "image"],
+                ['<td data-th="Caption">', "caption"],
+                ['<td data-th="Link">', "link"],
+                ['<td data-th="Source">', "source"]
+            ]
```

Comment:
This code looks similar to some code that you wrote earlier. If you are indeed using the same code in multiple places, I would recommend breaking it into a function.

Reply:
hmm, then I'll convert the object into a class, that should make it a bit easier to reuse code.

Reply:
If it is easier/relevant, you can transform into a function and just have these as subfunctions.
```javascript
function outer() {
         outer.inner = function(){}
}
```

Overall, I do not have any preference on whether it is an object, class, or function, so do whatever you think is best! This is certainly complex code.